scripts/fetch/map_data.R

```diff
@@ -0,0 +1,42 @@
+fetchTimestamp.map_data_states <- fetchTimestamp.map_data_counties <- vizlab::alwaysCurrent
+
+#' Gets data for state polygons.
+#' @param viz a vizlab object including a \code{spatial_metadata} parameter input
+#' @details 
+#' Depends on: \code{spatial_metadata} which must include:
+#' /describe{
+#'   \item{bbox}{numeric in xmin, ymin, xmax, ymax order}
+#'}
+#' and optionally: 
+#' /describe{
+#'   \item{crs}{valid crs for \pkg{sf}}
+#' }
+fetch.map_data_states <- function(viz = as.viz("map_data_states")){
+
+  sm <- spatial_meta(viz)
+  
+  saveRDS(get_map_data_sf(database = 'state', 
+                          crs = sm$crs, 
+                          within = sm["bbox_polygon"]), 
+          viz[['location']])
+  }
+
+#' Gets data for county polygons.
+fetch.map_data_counties <- function(viz = as.viz("map_data_counties")){
+  
+  sm <- spatial_meta(viz)
+  
+  saveRDS(get_map_data_sf(database = 'county', 
+                          crs = sm$crs, 
+                          within = sm["bbox_polygon"]), 
+          viz[['location']])
+}
+
+spatial_meta <- function(viz) {
+  deps <- readDepends(viz)
+  checkRequired(deps, "spatial_metadata")
+  spatial_meta <- deps[["spatial_metadata"]]
+  spatial_meta$bbox_polygon <- bbox_to_polygon(spatial_meta$bbox,
+                                               return_crs = spatial_meta$crs)
+  return(spatial_meta)
```

Comment:
Not sure I understand what this function does: it is called `spatial_meta`, takes the `spatial_metadata` from a viz, adds a field to it that is `sfc`, and returns it? 
Why not skip this function (and the call to it) and change the 
```r
within = bbox_to_polygon(...)
```
where the spatial_metadata is used to get the bbox. 
but see comment above about using viewbox_limits instead of spatial_metadata. Also, I had a question buried somewhere else that asked if we should be returning a polygon (instead of ylim xlim) from `viewbox_limits`. Returning the polybox here would make things easier.

Reply:
This was just to reduce duplicate code above. I was trying not to repeat the checking of depends in four duplicated fetchers. Given the first comment above, this can go away.